app/services/fees/calculator.go

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2021 LimeChain Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fees
+
+import (
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/limechain/hedera-eth-bridge-validator/app/domain/provider"
+	"github.com/limechain/hedera-eth-bridge-validator/app/helper"
+	"github.com/limechain/hedera-eth-bridge-validator/config"
+	"github.com/shopspring/decimal"
+	"math"
+	"math/big"
+)
+
+type FeeCalculator struct {
+	rateProvider  provider.ExchangeRateProvider
+	configuration config.Hedera
+}
+
+func NewFeeCalculator(rateProvider provider.ExchangeRateProvider, configuration config.Hedera) *FeeCalculator {
+	return &FeeCalculator{
+		rateProvider:  rateProvider,
+		configuration: configuration,
+	}
+}
+
+func (fc FeeCalculator) ValidateExecutionFee(strTransferFee string, transferAmount string, gasPrice string) (bool, error) {
+	bigTransferAmount, err := helper.ToBigInt(transferAmount)
+	if err != nil {
+		return false, InvalidTransferAmount
+	}
+
+	serviceFeePercent := new(big.Int).SetUint64(fc.configuration.Client.ServiceFeePercent)
+	bigServiceFee := new(big.Int).Mul(bigTransferAmount, serviceFeePercent)
+	bigServiceFee = new(big.Int).Div(bigServiceFee, new(big.Int).SetInt64(100))
```

Comment:
We should be charging service fee only on the transferred amount. At the moment we are charging a service fee for the txReimbursementFee as-well

We should do `(transferAmount - transferFee)*serviceFee/100` and not `transferAmount*serviceFee/100`